cmd/collection/main.go

```diff
@@ -373,7 +374,13 @@ func main() {
 		}).
 		// Epoch manager encapsulates and manages epoch-dependent engines as we
 		// transition between epochs
-		Component("epoch manager", func(_ cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+		Component("epoch manager", func(nodeBuilder cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+
+			workerPool := workerpool.New(4)
+			nodeBuilder.Component("epoch factory worker pool", func(_ cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
```

Comment:
To me, initializing the `workerPool` as a stand-alone `Component` seems a bit of a hacky approach: 
* Conceptually, I feel that the Worker pool belongs to the `VoteAggregator` and should be managed by it. All other worker threads are usually tied to the component. 

Having said this, I think it might require a noteworthy amount of work to update the code. Not sure it is worth it ... probably not as part of this PR anyway.